src/components/WeekCalendar.tsx

```diff
@@ -59,6 +60,28 @@ const WeekCalendar = ({ events, focusDate, onFocusDate }: WeekCalendarProps): JS
         };
     };
 
+    const getCampColors = (campId?: number) => {
```

Comment:
I would probably move the main logic of this function out into a separate module and call it something like `getContrastColor()`, so that you can use something like `const fg = getContrastColor(campaign.color)`.

Perhaps put it in the same module as `generateRandomColor()`, rename the module `colorUtils` and disregard my previuos comment about default exports.

Reply:
yes!